core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/FulltextSearchQueryUtils.scala

```diff
@@ -0,0 +1,151 @@
+package edu.uci.ics.texera.web.resource.dashboard
+
+import edu.uci.ics.texera.web.model.jooq.generated.Tables._
+import edu.uci.ics.texera.web.resource.dashboard.DashboardResource.SearchQueryParams
+import org.jooq.{Condition, Field, OrderField}
+import org.jooq.impl.DSL.{condition, noCondition}
+
+import java.sql.Timestamp
+import java.text.{ParseException, SimpleDateFormat}
+import java.util.concurrent.TimeUnit
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+
+object FulltextSearchQueryUtils {
+
+  def getOrderFields(
+      specificResourceType: String,
+      searchQueryParams: SearchQueryParams
+  ): List[OrderField[_]] = {
+    // Regex pattern to extract column name and order direction
+    val pattern = "(Name|CreateTime|EditTime)(Asc|Desc)".r
```

Comment:
those column names seem to be hard-coded. is there a way to avoid hardcoding them?

Reply:
no. they are from the frontend.